observableeditablebuffer.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import "fmt"
+
+type observableeditablebuffer struct {
```

Comment:
Go idiom: `ObservableEditableBuffer` goes in `observableeditablebuffer.go`. Given that another goal is to move this into `internal/file`, it would be good to write it under that assumption.